newsfragments/3337.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Fixed a rare TOC/TOU bug in `trio.lowlevel.Task.iter_await_frames` where
+the underlying coroutine has already been closed.
```

Comment:
Actually I don't think this is TOC/TOC? Isn't this just "`iter_await_frames` exits with an error when run on a finished task, which means in a multi-threaded system it is hard to use correctly"?

I'm probably missing things though.

Reply:
"When run on a task that has finished the underlying async computation", yeah.  It doesn't have to be multithreaded, you just need to hang onto basically any reference to the current task and (try to) iterate it's frames later.

I'm not convinced this is _useful_ to be clear, but it should return an empty list of frames ("there is no location") rather than raising an `AttributeError` imo.

Reply:
I was mistaken earlier, it's not actually a TOC/TOU issue at all.

Reply:
> I'm not convinced this is _useful_ to be clear, but it should return an empty list of frames ("there is no location") rather than raising an `AttributeError` imo.

I think this is better behavior for a tracing function, plus the docs don't even say not to run this on finished tasks!